app/src/main/java/com/kickstarter/ui/activities/compose/search/CategorySelectionSheet.kt

```diff
@@ -0,0 +1,310 @@
+                    ) {
+                        KSButton(
+                            onClickAction = {
+                                selectedCategories = selectedCategories.mapValues { false }
+                            },
+                            type = KSButtonType.Outlined,
+                            text = stringResource(R.string.Reset),
+                            isEnabled = !isLoading
+                        )
+                        KSButton(
+                            modifier = Modifier.weight(1f),
+                            onClickAction = { onApply(totalSelectedResults) },
+                            type = KSButtonType.Filled,
+                            text = "See $totalSelectedResults results",
+                            isEnabled = totalSelectedResults > 0 && !isLoading,
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun CategoryItemRow(
+    category: CategoryItem,
+    isSelected: Boolean,
+    onSelectionChange: (Boolean) -> Unit,
+) {
+    val backgroundDisabledColor = colors.backgroundDisabled
+    val dimensions: KSDimensions = KSTheme.dimensions
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .drawBehind {
+                drawLine(
+                    color = backgroundDisabledColor,
+                    start = Offset(0f, size.height),
+                    end = Offset(size.width, size.height),
+                    strokeWidth = dimensions.dividerThickness.toPx()
+                )
+            }
+            .padding(horizontal = dimensions.paddingLarge, vertical = dimensions.paddingMedium),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Row(
+            modifier = Modifier.weight(1f),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Text(
+                text = category.name,
+                style = KSTheme.typographyV2.headingLG
+            )
+
+            Box(
+                modifier = Modifier
+                    .padding(start = dimensions.paddingSmall)
+                    .background(
+                        color = if (isSelected) colors.borderAccentGreenSubtle.copy(alpha = 0.3f) else backgroundDisabledColor,
+                        shape = RoundedCornerShape(dimensions.radiusExtraSmall)
+                    )
+                    .padding(
+                        horizontal = dimensions.paddingSmall,
+                        vertical = dimensions.paddingXSmall
+                    )
+            ) {
+                Text(
+                    text = category.totalResults.toString(),
+                    style = KSTheme.typographyV2.headingSM,
+                    color = colors.kds_black
+                )
+            }
+        }
+
+        CustomSwitch(
+            checked = isSelected,
+            onCheckedChange = onSelectionChange,
+        )
+    }
+}
+
+@Composable
+fun CustomSwitch(
```

Comment:
The switch can be added to Design System as well,  on the "controls" category. Will add a follow up ticket for it :) 
<img width="382" alt="Screenshot 2025-03-03 at 11 37 43 AM" src="https://github.com/user-attachments/assets/0b3271e1-11b9-4362-8e96-133d21fc9450" />